native-image/wasm-javac/README.md

```diff
@@ -0,0 +1,47 @@
+# Compile `javac` to a Wasm Module
+
+This demo illustrates how to use the WebAssembly backend of GraalVM to compile the `javac` tool into a Wasm module that can then run on the command-line or in the browser.
+
+## Prerequisites
+
+This demo requires:
+
+1. An [Early Access Build](https://github.com/graalvm/oracle-graalvm-ea-builds) of Oracle GraalVM for JDK 25 or later.
+    For example, using SDKMAN!: `sdk install java 25.ea.15-graal`
+2. The [Binaryen toolchain](https://github.com/WebAssembly/binaryen) in version 119 or later and on the system path.
+    For example, using Homebrew: `brew install binaryen`
+
+
+## Run `javac` on Node
+
+1. Build the Wasm module:
+    ```bash
+    mvn -Pnative package
+    ```
+    Uses the [Native Build Tools](https://graalvm.github.io/native-build-tools/latest/index.html) to run the `native-image` executable from Maven.
+    This command generates a Wasm file and a corresponding JavaScript binding in the `target` directory.
+
+2. Compile a Java file:
+    ```bash
+    node target/javac.js HelloWasm.java
+    ```
+    The resulting `HelloWasm.class` file is placed next to `HelloWasm.java`.
+    This requires Node.js 22 or later.
+
+## Run `javac` in the browser
+
+1. Build the Wasm module:
+    ```bash
+    mvn -Pjavacweb package
+    ```
+    This command generates a Wasm file and a corresponding JavaScript binding in the `web` directory.
+
+2. Run a local web server in the `web` directory:
+    ```bash
+    cd web
+    jwebserver
+    ```
+    This will serve the `web` directory locally on port `8000`.
+
+3. Navigate to http://localhost:8000 to compile Java programs in the browser:
```

Comment:
Can we add some short summary for this demo  at the end? Or some links to read more?